src/emit/functions.rs

```diff
@@ -33,14 +33,13 @@ pub(super) fn emit_functions<'a, T: TargetRuntime<'a>>(
                 ns,
             );
 
-            let func_decl = if let Some(func) = bin.module.get_function(&cfg.name) {
+            let func_decl = if let Some(func) = bin.module.get_function(&cfg.original_name) {
                 // must not have a body yet
                 assert_eq!(func.get_first_basic_block(), None);
 
                 func
             } else {
-                bin.module
-                    .add_function(&cfg.name, ftype, Some(Linkage::Internal))
+                bin.module.add_function(&cfg.original_name, ftype, None)
```

Comment:
I removed the internal linkage on emitted functions and instead rely completely on the `internalize` function to mark internal functions as internal.